Marketplace/musicVisualizer/musicVisualizer.js

```diff
@@ -0,0 +1,293 @@
+                equipLeftProps.visible = true;
+                equipRightProps.visible = true;
+                Overlays.editOverlay(equipLeft, equipLeftProps);
+                Overlays.editOverlay(equipRight, equipRightProps);
+
+            }, 700);
+
+            // once particles are equipped tablet must be auto closed
+            button.editProperties({ isActive: false });
+            tablet.webEventReceived.disconnect(onWebEventReceived);
+            tablet.gotoHomeScreen();
+            Messages.sendLocalMessage("home", HMD.homeButtonID);
+            shown = false;
+        }
+
+        if (behavior === "nobehavior") {
+            var noBehaviorParticle = Entities.addEntity(effectJSON);
+            existingParticles.push(noBehaviorParticle);
+        }
+
+        // clears audio file selection
+        updateSelection("");
+    }
+
+    // to modify HTML page's listed filename
+    function updateSelection(filename) {
+        if (filename === "") {
+            audioFile = "";
+        }
+        tablet.emitScriptEvent(JSON.stringify({
+            "file": filename
+        }));
+    }
+
+    function onWebEventReceived(event) {
+        var htmlEvent = JSON.parse(event);
+        print("Event: " + JSON.stringify(htmlEvent));
+        // Handles particle button clicks to retrieve the effect JSON from musVisLib
+        if (htmlEvent.type === "click") {
+            var effectName = htmlEvent.data;
+            var micSync = htmlEvent.sync;
+            var file = audioFile;
+            var behavior = htmlEvent.behavior;
+            print("Audio file: " + file);
+            createParticle(effectName, micSync, behavior, file);
+        // Handles audio file browsing event
+        } else if (htmlEvent.type === "chooseAudioFile") {
+            if (htmlEvent.value !== CLEAR_SELECTION_TEXT) {
+                audioFile = Window.browse("Choose a .wav audio file", "", "*.wav");
+                if (audioFile !== null) {
+                    var filenameArr = audioFile.split("/");
+                    var filename = filenameArr[filenameArr.length - 1];
+                    updateSelection(filename);
+                } else {
+                    updateSelection("");
+                }
+            } else {
+                updateSelection("");
+            }
+            
+        // Resets saved audio file when contradiction present
+        } else if (htmlEvent.type === "contradiction") {
+            updateSelection("");
+        } else if (htmlEvent.type === "cleanUpParticles") {
+            cleanUpParticles();
+        }
+    }
+
+    function cleanUpParticles() {
+        print("Cleaning up particles");
+        
+        var getNearbyEntities = Entities.findEntities(MyAvatar.position, DELETE_RADIUS);
+        for (var i = 0; i < getNearbyEntities.length; i++) {
+            var nearbyEntity = getNearbyEntities[i];
+            var nearbyEntityProperties = Entities.getEntityProperties(nearbyEntity);
```

Comment:
Please only request the properties that you need when calling getEntityProperties